portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -81,16 +100,57 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     
+    UserService userService = UserServiceFactory.getUserService();
     String content = request.getParameter("content");
-    String author = request.getParameter("author");
+    User currentUser = getUserInfo(userService.getCurrentUser().getUserId());
     long timestamp = System.currentTimeMillis();
 
     Entity commentEntity = new Entity("Comment");
     commentEntity.setProperty("timestamp", timestamp);
     commentEntity.setProperty("content", content);
-    commentEntity.setProperty("author", author);
+    commentEntity.setProperty("author", currentUser.getName());
+    commentEntity.setProperty("image", currentUser.getImage());
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
   }
+
+  /**
+   * Returns the nickname of the user from id, if one doesn't exist, create a default account.
+   */
+  private User getUserInfo(String id) {
+
+    // Set up datastore.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    // Get entity of the user based on id.
+    Query query =
+      new Query("User")
+        .setFilter(new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, id));
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+    
+    // If user not in datastore, create new user.
+    if (entity == null) {
+      Entity userEntity = new Entity("User", id);
+      userEntity.setProperty("id", id);
+      
+      Random rand = new Random();
+      int number = rand.nextInt(9000) + 1000;
+      String name = "User" + Integer.toString(number);
+      userEntity.setProperty("name", name);
```

Comment:
Is there an easier way to do this? Are the 'ids' unique? In that case could the name be "User"+id ?

Reply:
Right now, uniqueness is not checked/ maintained. Using the id would be much easier though, thanks!